### PR TITLE
[9.0][MIG] account_payment_order: Migration script

### PR DESCRIPTION
**Pull request by sergio-teruel**
WIP
cc @tecnativa

### CODE REVIEW

File: account_payment_order/migrations/9.0.1.1.1/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    ],
+    'bank_payment_line': [
+        ('transit_move_line_id', None),
+    ],
+}
+
+
+def install_new_modules(cr):
+    sql = """
+    UPDATE ir_module_module SET state='to install' WHERE name in %s
```
Comment (pedrobaeza) [inline account_payment_order/migrations/9.0.1.1.1/pre-migration.py]: I think this is automatically handled by OpenUpgrade. Please check and remove if so. If not, please include the condition that they are in state='uninstalled' (they can be already installed in the source database).

File: account_payment_order/migrations/9.0.1.1.1/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+# account_payment_order
+to_install = [
+    'account_payment_mode',
+    'account_payment_partner',
+]
+
+table_renames = [
+    ('payment_order', 'account_payment_order'),
+    ('payment_line', 'account_payment_line'),
+    ]
```
Comment (pedrobaeza) [inline account_payment_order/migrations/9.0.1.1.1/pre-migration.py]: De-indent this line

File: account_payment_order/migrations/9.0.1.1.1/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+# © 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+# V9 modules that are not exist in v8 and are dependent of
```
Comment (pedrobaeza) [inline account_payment_order/migrations/9.0.1.1.1/pre-migration.py]: s/that are not exist/that don't exist

File: account_payment_order/migrations/9.0.1.1.1/pre-migration.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        openupgrade.rename_columns(env.cr, column_renames_payment_export)
+
+    if openupgrade.is_module_installed(
+            env.cr, 'account_banking_payment_transfer'):
+        openupgrade.rename_columns(env.cr, column_renames_payment_transfer)
+        openupgrade.update_module_names(
+            env.cr,
+            [('account_banking_payment_transfer', 'account_payment_order')],
+            merge_modules=True)
+    openupgrade.rename_tables(env.cr, table_renames)
```
Comment (pedrobaeza) [inline account_payment_order/migrations/9.0.1.1.1/pre-migration.py:99]: Please use a function that I have just discovered that is more complete (also handles attachments and so on) and that applies here, as we are not renaming to None, but to a still useful table: _rename_models_ - https://github.com/OCA/openupgradelib/blob/c723fa4ccf288184d163d9daa211c3dc1281c263/openupgradelib/openupgrade.py#L362

File: account_payment_order/migrations/9.0.1.1.1/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+# © 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+def map_payment_type(cr):
+    openupgrade.map_values(
+        cr,
+        'payment_order_type', 'payment_type',
```
Comment (pedrobaeza) [inline account_payment_order/migrations/9.0.1.1.1/post-migration.py]: Both fields exists in new module?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Travis is red

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Merging optimistically as OpenUpgrade's policy.

**Pull request merged**